src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -189,23 +177,23 @@ protected void AssertSelf ()
 			if (!request.RequestUri.IsAbsoluteUri)
 				throw new ArgumentException ("Must represent an absolute URI", "request");
 
-			java_url = new URL (request.RequestUri.ToString ());
-			java_connection = java_url.OpenConnection () as HttpURLConnection;
-			HttpURLConnection httpConnection = await SetupRequestInternal (request, java_connection).ConfigureAwait (false);
-					return await ProcessRequest (request, httpConnection, cancellationToken);
+			var java_url = new URL (request.RequestUri.ToString ());
+			var java_connection = java_url.OpenConnection ();
+			HttpURLConnection httpConnection = SetupRequestInternal (request, java_connection);
+			return ProcessRequest (request, java_url, httpConnection, cancellationToken);
 		}
 		
-		Task <HttpResponseMessage> ProcessRequest (HttpRequestMessage request, HttpURLConnection httpConnection, CancellationToken cancellationToken)
+		Task <HttpResponseMessage> ProcessRequest (HttpRequestMessage request, URL javaUrl, HttpURLConnection httpConnection, CancellationToken cancellationToken)
 		{
 			cancellationToken.ThrowIfCancellationRequested ();
 			httpConnection.InstanceFollowRedirects = AllowAutoRedirect;
 			RequestedAuthentication = null;
 			ProxyAuthenticationRequested = false;
-			
-			return DoProcessRequest (request, httpConnection, cancellationToken);
+
+			return DoProcessRequest(request, javaUrl, httpConnection, cancellationToken);
```

Comment:
Please follow the [Mono Code Style Guidelines](http://www.mono-project.com/community/contributing/coding-guidelines/#style-guidelines) (there should be a space between method name and the opening parenthesis)